@here/olp-sdk-dataservice-write/lib/client/VersionedLayerClient.ts

```diff
@@ -509,4 +512,256 @@ export class VersionedLayerClient {
+            return Promise.reject(
+                new Error("Please set layerId to the UploadBlobRequest")
+            );
+        }
+
+        const contentEncoding = request.getContentEncoding() as
+            | BlobApi.ContentEncodingEnum
+            | undefined;
+        const contentType = request.getContentType();
+        if (!contentType) {
+            return Promise.reject(
+                new Error("Please set contentType to the UploadBlobRequest")
+            );
+        }
+
+        const data = request.getData();
+        if (!data) {
+            return Promise.reject(
+                new Error("Please set data to the UploadBlobRequest")
+            );
+        }
+
+        let dataHandle = request.getDataHandle();
+
+        if (!dataHandle) {
+            /**
+             * If datahandle was not provided, we generate the UUID and use it as datahandle.
+             * A retry logic in case datahandle is already present is a loop of 3 tries
+             * and each time we generate a new UUID and retry.
+             *
+             * If all thies was not success, the method rejects the promise with Error.
+             */
+            const timesToGenerate = 3;
+            const checkDataExistsRequest = new CheckDataExistsRequest().withLayerId(
+                layerId
+            );
+            if (billingTag) {
+                checkDataExistsRequest.withBillingTag(billingTag);
+            }
+
+            for (
+                let generatingCount = 0;
+                generatingCount < timesToGenerate;
+                generatingCount++
+            ) {
+                const generatedDatahandle = Uuid.create();
+                const dataExist = await this.checkDataExists(
+                    checkDataExistsRequest.withDataHandle(generatedDatahandle)
+                ).catch(async (response: Response) => {
+                    if (response.status === STATUS_CODES.NOT_FOUND) {
+                        dataHandle = generatedDatahandle;
+                        return Promise.resolve(false);
+                    }
+
+                    return Promise.reject(response);
+                });
+
+                if (!dataExist) {
+                    break;
+                }
+            }
+
+            if (!dataHandle) {
+                return Promise.reject(
+                    new Error("Please set DataHandle to the UploadBlobRequest")
+                );
+            }
+        }
+
+        const blobBuilder = await RequestFactory.create(
+            "blob",
+            "v1",
+            this.params.settings,
+            this.params.catalogHrn,
+            abortSignal
+        );
+
+        const multipartUploadParams: BlobApi.MultipartUploadMetadata = {
+            contentType
+        };
+
+        if (contentEncoding !== undefined) {
+            multipartUploadParams.contentEncoding = contentEncoding;
+        }
+
+        const multipartUploadMeta = (await BlobApi.startMultipartUpload(
```

Comment:
But why are you always doing a multipart upload? In case the data is below 50Mb you just upload directly via putBlob, see https://developer.here.com/documentation/data-api/api-reference-blob.html#operation%2FputBlob.
Always using multipart upload is quite expensive for small data.

Reply:
Good point, updated. We're using multipart only if size more than 50Mb from now. Thanks.